synapse/handlers/register.py

```diff
@@ -513,6 +513,34 @@ def get_or_register_3pid_guest(self, medium, address, inviter_user_id):
 
     @defer.inlineCallbacks
     def _join_user_to_room(self, requester, room_identifier):
+
+        # try to create the room if we're the first user on the server
+        if self.hs.config.autocreate_auto_join_rooms:
+            count = yield self.store.count_all_users()
+            if count == 1 and RoomAlias.is_valid(room_identifier):
+                room_creation_handler = self.hs.get_room_creation_handler()
+                info = yield room_creation_handler.create_room(
+                    requester,
+                    config={
+                        "preset": "public_chat",
+                    },
+                    ratelimit=False,
+                )
+                room_id = info["room_id"]
+
+                directory_handler = self.hs.get_handlers().directory_handler
```

Comment:
can we not do the alias bits by passing `room_alias_name` to `create_room` ?

Reply:
oh, possibly.